pkg/components/azure-arc-onboarding/component_test.go

```diff
@@ -0,0 +1,196 @@
+	c := azurearconboarding.NewConfig()
+
+	config := `
+component azure-arc-onboarding {
+  application_client_id = ""
+  tenant_id             = ""
+  application_password  = ""
+  resource_group        = ""
+  cluster_name          = ""
+}
+`
+
+	body, diagnostics := util.GetComponentBody(config, azurearconboarding.Name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Error getting component body: %v", diagnostics)
+	}
+
+	if diagnostics := c.LoadConfig(body, &hcl.EvalContext{}); !diagnostics.HasErrors() {
+		t.Fatalf("Config with empty fields should return error")
+	}
+}
+
+func TestConfig(t *testing.T) { //nolint:funlen
+	c := azurearconboarding.NewConfig()
+
+	tests := []struct {
+		desc    string
+		config  string
+		wantErr bool
+	}{
+		{
+			desc: "Valid config",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+		},
+		{
+			desc: "Empty config",
+			config: `
+component azure-arc-onboarding {}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty application_client_id",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = ""
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty tenant_id",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = ""
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty application_password",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "tenant-id"
+  application_password  = ""
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty resource_group",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = ""
+  cluster_name          = "foobar"
+}
+			`,
+			wantErr: true,
+		},
+		{
+			desc: "Empty cluster_name",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = ""
+}
+			`,
+			wantErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		body, diagnostics := util.GetComponentBody(test.config, azurearconboarding.Name)
```

Comment:
When you to test table, each case should run in it's own `t.Run()` with `t.Parallel()`, so single test case can be executed and that you know exactly which scenario fails.

Reply:
I highly recommend taking similar approach as in for example https://github.com/newrelic/newrelic-infra-operator/blob/46004321ee2bc1c11c0cedb55549e4cbc75a7655/internal/mutator/pod/agent/injector_test.go#L45-L148